### PR TITLE
Modularize event Analytics live tail and fix bug

### PR DESCRIPTION
**Pull request by kavithacm**
Signed-off-by: Kavitha Conjeevaram Mohan <mohakavi@amazon.com>

### Description

1. Modularizing Live Tail so that it can be used by any plugin under Observability 
2. Fixed bug- Live tail runs in background if the page is moved using breadcrumbs.

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: dashboards-observability/public/components/common/live_tail/live_tail_button.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    isLiveTailOn,
+    isLiveTailPopoverOpen,
+    setIsLiveTailPopoverOpen,
+    liveTailName,
+    dataTestSubj,
+}: LiveTailProps) => {
+
+    const liveButton = useMemo(() => {
+        return (
+          <EuiButton
```
Comment (joshuali925) [inline dashboards-observability/public/components/common/live_tail/live_tail_button.tsx:23]: np: inconsistent indentation, run a formatter?

File: dashboards-observability/public/components/common/live_tail/live_tail_button.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            >
+            Stop
+            </EuiButton>
+        )
+    };
+    return stopButton();
+};
+
+export const sleep = (milliseconds: number | undefined) => {
+    return new Promise((resolve) => setTimeout(resolve, milliseconds));
```
Comment (joshuali925) [inline dashboards-observability/public/components/common/live_tail/live_tail_button.tsx:57]: where is this used?

File: dashboards-observability/public/components/explorer/explorer.tsx [hunk 3]
```diff
@@ -1115,81 +1058,29 @@ export const Explorer = ({
...
-    <EuiContextMenuItem
-      key="1h"
-      onClick={async () => {
-        liveTailLoop('1h', 'now-1h', 'now', 60000 * 60);
-      }}
-    >
-      1h
-    </EuiContextMenuItem>,
+  //stop live tail if the page is moved using breadcrumbs
+  var lastUrl = location.href;
```
Comment (joshuali925) [inline dashboards-observability/public/components/explorer/explorer.tsx:1062]: use let instead of var

File: dashboards-observability/.cypress/integration/event_analytics.spec.js [hunk 4]
```diff
@@ -394,7 +394,7 @@ describe('Switch on and off livetail', () => {
     cy.get('[data-test-subj="searchAutocompleteTextArea"]').type(TEST_QUERIES[1].query);
 
     cy.get('[data-test-subj=eventLiveTail]').click();
-    cy.get('[data-test-subj=eventLiveTail__delay10]').click();
+    cy.get('[data-test-subj=10s]').click();
```
Comment (joshuali925) [inline dashboards-observability/.cypress/integration/event_analytics.spec.js:397]: any ways to make the test id naming more specific?

File: dashboards-observability/public/components/explorer/explorer.tsx [hunk 5]
```diff
@@ -323,11 +323,16 @@ export const Explorer = ({
...
+      endTime = curQuery![SELECTED_DATE_RANGE][1];
+    }
+
     // compose final query
     const finalQuery = composeFinalQuery(
       curQuery,
-      curQuery![SELECTED_DATE_RANGE][0],
-      curQuery![SELECTED_DATE_RANGE][1],
+      startTime,
+      endTime,
```
Comment (joshuali925) [inline dashboards-observability/public/components/explorer/explorer.tsx:335]: is it possible for them to be empty strings? i don't think composeFinalQuery handles empty strings

File: dashboards-observability/public/components/explorer/explorer.tsx [hunk 6]
```diff
@@ -445,7 +405,7 @@ export const Explorer = ({
     if (objectId) {
       updateTabData(objectId);
     } else {
-      fetchData();
+      fetchData('', '');
```
Comment (joshuali925) [inline dashboards-observability/public/components/explorer/explorer.tsx:408]: instead of passing empty strings maybe make parameters optional?

File: dashboards-observability/.cypress/integration/event_analytics.spec.js (hunk 4 above)
Comment (davidcui1225) [reply dashboards-observability/.cypress/integration/event_analytics.spec.js:397]: +1 - if we can't find an intuitive ID to  call this by, could we extract this line to a function that describes the action? e.g `clickLivetailPlayButton()`

File: dashboards-observability/public/components/common/live_tail/live_tail_button.tsx (hunk 1 above)
Comment (kavithacm) [reply dashboards-observability/public/components/common/live_tail/live_tail_button.tsx:23]: yeah okay

File: dashboards-observability/public/components/explorer/explorer.tsx (hunk 3 above)
Comment (kavithacm) [reply dashboards-observability/public/components/explorer/explorer.tsx:1062]: yeah okay

File: dashboards-observability/.cypress/integration/event_analytics.spec.js (hunk 4 above)
Comment (kavithacm) [reply dashboards-observability/.cypress/integration/event_analytics.spec.js:397]: i'll try to make it more specific, if not i'll try the function method too

File: dashboards-observability/public/components/explorer/explorer.tsx (hunk 5 above)
Comment (kavithacm) [reply dashboards-observability/public/components/explorer/explorer.tsx:335]: ohh i have assigned its value before composeFinalQuery is called

File: dashboards-observability/public/components/common/live_tail/live_tail_button.tsx [hunk 7]
```diff
@@ -5,54 +5,51 @@
...
+import { LiveTailProps } from "common/types/explorer";
 
 //Live Tail Button
 export const LiveTailButton = ({
-    isLiveTailOn,
-    isLiveTailPopoverOpen,
-    setIsLiveTailPopoverOpen,
-    liveTailName,
-    dataTestSubj,
+	isLiveTailOn,
```
Comment (joshuali925) [inline dashboards-observability/public/components/common/live_tail/live_tail_button.tsx:14]: are you not using prettier as formatter? i don't think we use tabs

**Review by joshuali925 — DISMISSED** [review]

File: dashboards-observability/public/components/common/live_tail/live_tail_button.test.tsx [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+  describe('Live tail button', () => {
+    configure({ adapter: new Adapter() });
+  
+    it('starts live tail with 5s interval', async () => {
+      const setIsLiveTailPopoverOpen = jest.fn();
+
+      const wrapper = mount(
+        <LiveTailButton 
+        isLiveTailOn={true}
```
Comment (mengweieric) [inline dashboards-observability/public/components/common/live_tail/live_tail_button.test.tsx:20]: Minor: indentations

**Review by davidcui1225 — APPROVED** [review]

File: dashboards-observability/public/components/common/live_tail/live_tail_button.tsx (hunk 7 above)
Comment (kavithacm) [reply dashboards-observability/public/components/common/live_tail/live_tail_button.tsx]: i did format it but unfortunately it wasn't using prettier. now changed it to prettier

**Review by joshuali925 — APPROVED** [review]

File: dashboards-observability/public/components/explorer/explorer.tsx [hunk 9]
```diff
@@ -1250,6 +1130,8 @@ export const Explorer = ({
           tabId={tabId}
           baseQuery={appBaseQuery}
           stopLive={stopLive}
+          setIsLiveTailPopoverOpen={setIsLiveTailPopoverOpen}
```
Comment (mengweieric) [inline dashboards-observability/public/components/explorer/explorer.tsx:1143]: I'm thinking also like how about removing all these live tail related props and pass them directly to your live tail component, and then you pass the live tail component to search? Since all these props are not related to search component. Keep adding props that not related to search would make it unmanageable.

File: dashboards-observability/public/components/common/live_tail/live_tail_button.test.tsx (hunk 8 above)
Comment (mengweieric) [reply dashboards-observability/public/components/common/live_tail/live_tail_button.test.tsx:20]: ignore these if you've already pushed formatted changes

File: dashboards-observability/public/components/explorer/explorer.tsx (hunk 9 above)
Comment (kavithacm) [reply dashboards-observability/public/components/explorer/explorer.tsx]: i'll create another PR to include those changes

**Review by mengweieric — APPROVED** [review]

File: dashboards-observability/public/components/common/live_tail/__tests__/live_tail_button.test.tsx [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { configure, mount, shallow } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import React from 'react';
+import { LiveTailButton, StopLiveButton } from '../live_tail_button';
+import { waitFor } from '@testing-library/dom';
+
+  describe('Live tail button', () => {
```
Comment (davidcui1225) [inline dashboards-observability/public/components/common/live_tail/__tests__/live_tail_button.test.tsx:12]: I believe the `describe` should not be indented, can we reduce the indent here?

**Review by davidcui1225 — APPROVED** [review]
LGTM- left 1 minor comment

File: dashboards-observability/public/components/common/live_tail/__tests__/live_tail_button.test.tsx (hunk 10 above)
Comment (kavithacm) [reply dashboards-observability/public/components/common/live_tail/__tests__/live_tail_button.test.tsx:12]: yeah okay, one more PR is yet to come, will include the change in that
